modules/zarr/src/lib/load-zarr.ts

```diff
@@ -1,47 +1,24 @@
-import type { ZarrArray } from 'zarr';
-import { loadMultiscales, guessTileSize } from './utils/zarr-utils';
+import type { Store } from 'zarr/types/storage/types';
+import { loadMultiscales, guessTileSize, guessLabels, normalizeStore, validLabels } from './utils';
 import ZarrPixelSource from './zarr-pixel-source';
-import type { Labels } from '../types';
 
-interface Channel {
-  active: boolean;
-  color: string;
-  label: string;
-  window: {
-    min?: number;
-    max?: number;
-    start: number;
-    end: number;
-  };
+interface ZarrOptions {
+  labels?: string[];
 }
 
-interface Omero {
-  channels: Channel[];
-  rdefs: {
-    defaultT?: number;
-    defaultZ?: number;
-    model: string;
-  };
-  name?: string;
-}
-
-interface Multiscale {
-  datasets: { path: string }[];
-  version?: string;
-}
-
-export interface RootAttrs {
-  omero: Omero;
-  multiscales: Multiscale[];
-}
-
-export async function loadZarr(store: ZarrArray['store']) {
+export async function loadZarr(root: string | Store, options: ZarrOptions = {}) {
+  const store = normalizeStore(root);
   const { data, rootAttrs } = await loadMultiscales(store);
-  const labels = ['t', 'c', 'z', 'y', 'x'] as Labels<['t', 'c', 'z']>;
   const tileSize = guessTileSize(data[0]);
+  const labels = options.labels ?? guessLabels(rootAttrs);
```

Comment:
If no labels are provided, they are inferred (if possible) from the `rootAttrs`. For now, `guessLabels` only infers if OME-Zarr.